### PR TITLE
[new release] git, git-mirage, git-paf and git-unix (3.11.0)

### PR DESCRIPTION
**Pull request by dinosaure**
Git format and protocol in pure OCaml

- Project page: <a href="https://github.com/mirage/ocaml-git">https://github.com/mirage/ocaml-git</a>
- Documentation: <a href="https://mirage.github.io/ocaml-git/">https://mirage.github.io/ocaml-git/</a>

##### CHANGES:

- Limit the stream between the incoming flow and the PACK analyzer
  It will bind the analyze with the I/O throughput (@dinosaure, mirage/ocaml-git#608)
- Remove OCaml 5.0.0 warnings (@samoht, mirage/ocaml-git#609)

### CODE REVIEW

**Comment by kit-ty-kate** [comment]
The tests fail on macOS. Is that expected?

**Comment by dinosaure** [comment]
It seems related to the `ca-certs` issue:
```
# +  fetch: internal error, uncaught exception:
# +         Invalid_argument("failed to load trust anchors: ca-certs: empty trust anchors.\nPlease report an issue at https://github.com/mirage/ca-certs, including:\n- the output of uname -s\n- the distribution you use\n- the location of default trust anchors (if known)\n")
# +         Raised at Stdlib.invalid_arg in file "stdlib.ml", line 30, characters 20-45
# +         Called from Dns_client_lwt.Transport.decode_resolv_conf in file "lwt/client/dns_client_lwt.ml", line 154, characters 24-40
# +         Called from Dns_client_lwt.Transport.resolv_conf in file "lwt/client/dns_client_lwt.ml", line 175, characters 8-33
# +         Called from Dns_client_lwt.Transport.create in file "lwt/client/dns_client_lwt.ml", line 227, characters 14-28
# +         Called from Dns_client.Make.create in file "client/dns_client.ml", line 227, characters 18-62
# +         Called from Happy_eyeballs_lwt.create in file "lwt/happy_eyeballs_lwt.ml", line 132, characters 70-94
# +         Called from Dune__exe__Fetch.main.(fun) in file "bin/guit/fetch.ml", line 78, characters 23-51
# +         Called from Lwt.Sequential_composition.bind.create_result_promise_and_callback_if_deferred.callback in file "src/core/lwt.ml", line 1849, characters 23-26
# +         Re-raised at Lwt.Miscellaneous.poll in file "src/core/lwt.ml", line 3077, characters 20-29
# +         Called from Lwt_main.run.run_loop in file "src/unix/lwt_main.ml", line 31, characters 10-20
# +         Called from Lwt_main.run in file "src/unix/lwt_main.ml", line 118, characters 8-13
# +         Re-raised at Lwt_main.run in file "src/unix/lwt_main.ml", line 124, characters 4-13
# +         Called from Dune__exe__Fetch.main in file "bin/guit/fetch.ml", line 149, characters 8-65
# +         Called from Cmdliner_term.app.(fun) in file "cmdliner_term.ml", line 24, characters 19-24
# +         Called from Cmdliner_eval.run_parser in file "cmdliner_eval.ml", line 34, characters 37-44
# +  [125]
```

**Comment by samoht** [comment]
That's https://github.com/mirage/ca-certs/issues/20

**Comment by kit-ty-kate** [comment]
> It seems related to the `ca-certs` issue:

In that case it needs to be constrainted so that the tests do not run on macOS

File: packages/git-unix/git-unix.3.11.0/opam [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+  "mirage-crypto-rng" {>= "0.8.8" & with-test}
+  "ptime"
+  "mimic"
+  "ca-certs-nss" {>= "3.60"}
+  "tls" {>= "0.14.0"}
+  "tls-mirage" {>= "0.14.0"}
+]
+build: [
+  ["dune" "build" "-p" name "-j" jobs]
+  ["dune" "runtest" "-p" name "-j" jobs "--no-buffer"] {with-test}
```
Comment (hannesm) [inline packages/git-unix/git-unix.3.11.0/opam:55]:
```suggestion
  ["dune" "runtest" "-p" name "-j" jobs "--no-buffer"] {with-test & os != "macos"} # the opam sandbox on macos leads to test failures (ocaml/opam#4389)
```

**Comment by kit-ty-kate** [comment]
All green after https://github.com/ocaml/opam-repository/pull/22925. Thanks!

**Pull request merged**